src/PatchOrchestrationApplication/NodeAgentNTService/src/Manager/WindowsUpdateManager.cs

```diff
@@ -299,13 +375,14 @@ private bool HandleWUOperationStates(CancellationToken cancellationToken)
                             OperationResultCode downloadResult = DownloadUpdates(cancellationToken);
                             reschedule = (downloadResult != OperationResultCode.orcSucceeded ? true : reschedule);
 
-                            WindowsUpdateOperationResult searchAndDownloadResult = this._operationResultFormatter.FormatSearchAndDownloadResult(downloadResult, this._wuCollectionWrapper);
+                            WindowsUpdateOperationResult searchAndDownloadResult = this._operationResultFormatter.FormatSearchAndDownloadResult(downloadResult, this._wuCollectionWrapper, this.lastUpdateOperationStartTimeStamp);
                             _eventSource.InfoMessage("Search and download result: {0}", searchAndDownloadResult);
 
                             this._nodeAgentSfUtility.UpdateSearchAndDownloadStatus(NodeAgentSfUtilityExitCodes.DownloadCompleted, searchAndDownloadResult, utilityTaskTimeOut);
 
                             string WUDownloadComplete = string.Format("Windows updates downloaded, waiting for installation approval from Repair Manager.");
                             this._nodeAgentSfUtility.ReportHealth(WUOperationStatus, WUDownloadComplete, HealthState.Ok, -1, TimeSpan.FromMinutes(this._serviceSettings.OperationTimeOutInMinutes));
+                            this.UpdateLastOperationStartTimeStamp(DateTime.UtcNow);
 
```

Comment:
You should move this to after installation approval

Reply:
Below as well

---
In reply to: [282396732](https://github.com/microsoft/Service-Fabric-POA/pull/15#discussion_r282396732) [](ancestors = 282396732)

Reply:
yes..done